src/builder.rs

```diff
@@ -209,10 +287,10 @@ impl<'a> RtpPacketBuilder<'a> {
             write_index += payload.len();
         }
 
-        if self.padded && (write_index & 0x3) != 0 {
+        if self.padded.is_padded() {
```

Comment:
You can't simplify this check.
Imagine this scenario:  
- Set your padding to `Pad::round_to(4)`
- Set the payload of the packet with a length of a multiple of 4.
This will result in setting the padding flag even though we've appended zero bytes of padding.
This will cause an invalid packet.

I would suggest a `should_pad(write_index)` method here.

Reply:
Thanks!  I'll get this fixed.